ios-trivia-game/ViewControllers/GameOptionsViewController.swift

```diff
@@ -10,17 +10,62 @@ import UIKit
 
 class GameOptionsViewController: UIViewController {
 
+    @IBOutlet weak var categoryPicker: UIPickerView!
+    @IBOutlet weak var numOfQuestionsPicker: UIPickerView!
+    
+    var categoryPickerData: [String] = [String]()
+    var numOfQuestionsPickerData: [String] = [String]()
+    
+    var numOfQuestions: Int?
+    
     override func viewDidLoad() {
         super.viewDidLoad()
 
         // Do any additional setup after loading the view.
+        getCategories()
+    }
+    
+    func getCategories() {
+        JServiceClient.instance.categories(count: 10, offset: 10, success: {(result: Any?) in
+            //print("my result : \(result)")
+            let triviaCategories = self.convertWithArray(dictionaries: result as! [NSDictionary])
+            for category in triviaCategories {
+                self.categoryPickerData.append(category.title!)
+            }
+            self.setupPickerData()
+            }, failure: { (error: Error?) in
+                    print("Error: \(error?.localizedDescription)")
+            })
+    }
+    
+    func convertWithArray(dictionaries: [NSDictionary]) -> [TriviaCategory] {
+        var categories = [TriviaCategory]()
+        
+        for dictionary in dictionaries {
+            let category = TriviaCategory(dictionary: dictionary)
+            categories.append(category)
+        }
+        
+        return categories
+    }
+    
+    func setupPickerData() {
+        categoryPicker.delegate = self
+        categoryPicker.dataSource = self
+        numOfQuestionsPicker.delegate = self
+        numOfQuestionsPicker.dataSource = self
+        self.numOfQuestionsPickerData = ["1", "2", "3", "4", "5", "6", "7", "8", "9", "10"]
     }
 
     override func didReceiveMemoryWarning() {
         super.didReceiveMemoryWarning()
         // Dispose of any resources that can be recreated.
     }
     
+    @IBAction func onBackClicked(_ sender: Any) {
+        dismiss(animated: true, completion: nil)
+        _ = self.navigationController?.popViewController(animated: true)
```

Comment:
Would the compiler complain if the function is invoked without assigning it to _? Interesting... Usually I use the _ only if it's part of a multi-part return and I need the other parts.